src/Buffer.sol

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Cartesi Pte. Ltd.
+
+// SPDX-License-Identifier: Apache-2.0
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+/// @dev This file is generated from helper_scripts/generate_AccessLogs.sh, one should not modify the content directly
+
+pragma solidity ^0.8.0;
+
+import "./Memory.sol";
+
+library Buffer {
+    using Buffer for Buffer.Context;
+    using Memory for Memory.AlignedSize;
+
+    struct Context {
+        bytes data;
+        uint256 offset;
+    }
+
+    function consumeBytes8(Buffer.Context memory buffer)
+        internal
+        pure
+        returns (bytes8)
+    {
+        bytes8 b8 = bytes8(buffer.peekBytes32());
+        buffer.offset += 8;
+
+        return b8;
+    }
+
+    function consumeBytes32(Buffer.Context memory buffer)
+        internal
+        pure
+        returns (bytes32)
+    {
+        bytes32 b32 = buffer.peekBytes32();
+        buffer.offset += 32;
+
+        return b32;
+    }
+
+    function peekBytes32(Buffer.Context memory buffer)
+        internal
+        pure
+        returns (bytes32)
+    {
+        bytes32 temp;
+        bytes memory data = buffer.data;
+        uint256 offset = buffer.offset;
+
+        // Get 32 bytes from data
+        assembly {
+            temp := mload(add(data, add(offset, 32)))
```

Comment:
The `32` is added because we are accessing a `byte` array. And an array in solidity always starts with its length which is a 32 byte-long variable. I'll put this in document.